contracts/abstracts/ListingManager.sol

```diff
@@ -58,6 +58,9 @@ abstract contract ListingManager is Controlable, IERC721ReceiverUpgradeable {
     Listing memory listing = _listings[listingId];
     require(listing.buyTimestamp == 0, 'ListingManager: Listing already sold');
     require(_listings[listingId].salePrice > 0, 'ListingManager: Sell price must be above zero or listing does not exist');
+    require(!_isBlacklistedUser(_listings[listingId].seller), 'ListingManager: Seller is blacklisted');
```

Comment:
please use the local memory value, to not read from storage again something we already assigned to memory